components/brave_ads/core/internal/serving/permission_rules/permission_rules.h

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ADS_CORE_INTERNAL_SERVING_PERMISSION_RULES_PERMISSION_RULES_H_
+#define BRAVE_COMPONENTS_BRAVE_ADS_CORE_INTERNAL_SERVING_PERMISSION_RULES_PERMISSION_RULES_H_
+
+namespace brave_ads {
+
+bool ShouldAllowFullScreenMode();
+bool ShouldAllowIssuers();
+bool ShouldAllowCommandLine();
+bool ShouldAllowConfirmationTokens();
+bool ShouldAllowUserActivity();
+bool ShouldAllowSearchResultAdsPerDay();
+bool ShouldAllowSearchResultAdsPerHour();
+bool ShouldAllowNewTabPageAdsPerDay();
+bool ShouldAllowNewTabPageAdMinimumWaitTime();
+bool ShouldAllowNewTabPageAdsPerHour();
+bool ShouldAllowNotificationAdsPerHour();
+bool ShouldAllowNotificationAdsPerDay();
+bool ShouldAllowNotificationAdMinimumWaitTime();
+bool ShouldAllowNetworkConnection();
+bool ShouldAllowMedia();
+bool ShouldAllowDoNotDisturb();
+bool ShouldAllowAllowNotifications();
+bool ShouldAllowCatalog();
+bool ShouldAllowInlineContentAdsPerDay();
+bool ShouldAllowInlineContentAdsPerHour();
+bool ShouldAllowPromotedContentAdsPerDay();
+bool ShouldAllowPromotedContentAdsPerHour();
+bool ShouldAllowBrowserIsActive();
```

Comment:
It would be great to move these permissions back to their own files (albeit as the much improved free functions).

Reply:
For the very small binary bloat, I would always favor readability, and loose coupling over these kinds of optimizations.

Reply:
Unless, of course, you have some measurements, so we can make sure there are no bottlenecks

Reply:
We use smaller files to stop the temptation for functions and classes to bind more tightly in the same file because they can access common but private knowledge. Breaking large files apart removes the temptation, breaks unnecessary dependencies, and promotes looser coupling. We are still on that journey, but we will get there.